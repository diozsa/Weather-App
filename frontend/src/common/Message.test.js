import React from "react";
import { render } from "@testing-library/react";
import Message from "./Message";

it("renders without crashing", function () {
  render(<Message />);
});

it("matches snapshot for danger", function () {
  let messages = ["Broken fragment", "This needs repairs"];
  const { asFragment } = render(<Message type="danger" messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot for success", function () {
  let messages = ["Everything runs smoothly!"];
  const { asFragment } = render(<Message type="success" messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});
