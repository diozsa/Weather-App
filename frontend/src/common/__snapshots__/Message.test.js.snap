// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for danger 1`] = `
<DocumentFragment>
  <div>
    <div
      class="fade alert alert-danger show"
      role="alert"
    >
      <p
        class="mb-0 small"
      >
        Broken fragment
      </p>
      <p
        class="mb-0 small"
      >
        This needs repairs
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for success 1`] = `
<DocumentFragment>
  <div>
    <div
      class="fade alert alert-success show"
      role="alert"
    >
      <p
        class="mb-0 small"
      >
        Everything runs smoothly!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
